packages/nixos/gpu.nix

```diff
@@ -10,6 +10,68 @@
 
 let
   cfg = config.contrast.gpu;
+
+  nvidiaPackage =
+    (
+      (config.boot.kernelPackages.nvidiaPackages.mkDriver {
+        # TODO(msanft): Investigate why the latest version breaks
+        # GPU containers.
+        version = "550.90.07";
+        sha256_64bit = "sha256-Uaz1edWpiE9XOh0/Ui5/r6XnhB4iqc7AtLvq4xsLlzM=";
+        sha256_aarch64 = "sha256-uJa3auRlMHr8WyacQL2MyyeebqfT7K6VU0qR7LGXFXI=";
+        openSha256 = "sha256-VLmh7eH0xhEu/AK+Osb9vtqAFni+lx84P/bo4ZgCqj8=";
+        settingsSha256 = "sha256-sX9dHEp9zH9t3RWp727lLCeJLo8QRAGhVb8iN6eX49g=";
+        persistencedSha256 = "sha256-qe8e1Nxla7F0U88AbnOZm6cHxo57pnLCqtjdvOvq9jk=";
+      }).override
+      {
+        disable32Bit = true;
+      }
+    ).overrideAttrs
+      (oldAttrs: {
+        # We strip the driver package by its dependencies on desktop software like Wayland and X11.
+        # For server use-cases, we shouldn't need these. The Mesa (and thus Perl) and libGL dependencies are dropped
+        # too, as GPU workloads will likely be AI-related and not graphical. The libdrm dependency is dropped as well,
+        # as we're probably not going to be watching Netflix on the servers.
+        # Source: https://github.com/NixOS/nixpkgs/blob/eac1633a086e8e109e00ce58c0b47721da1dbdfd/pkgs/os-specific/linux/nvidia-x11/generic.nix#L100C3-L114C6
+        libPath = lib.makeLibraryPath (
+          with pkgs;
+          [
+            zlib
+            stdenv.cc.cc
+            openssl
+            dbus # for nvidia-powerd
+          ]
+        );
+
+        # Hack to pass the "right" (i.e. the overridden) version of the nvidia driver to the persistenced.
+        # Looking at the package definition, it _should_ already do so, but it doesn't.
+        # So for now, override all occurences of `nvidia_x11` in the persistenced package "manually".
+        # We can't do an `override` on persistenced itself unfortunately, as it's call site doesn't allow this:
+        # https://github.com/NixOS/nixpkgs/blob/4d2418ebbfb107485b44aaa1b2909409322d9061/pkgs/os-specific/linux/nvidia-x11/generic.nix#L260
+        # TODO(msanft): Clarify with upstream why that is the case.
+        passthru = oldAttrs.passthru // {
+          persistenced = oldAttrs.passthru.persistenced.overrideAttrs (oldAttrs: {
+            inherit (nvidiaPackage) version makeFlags;
+            src = oldAttrs.src // {
+              rev = nvidiaPackage.version;
+            };
+
+            postFixup = ''
+              # Save a copy of persistenced for mounting in containers
```

Comment:
Why do we need the origBin?

Reply:
We probably don't. But I would like to not diverge from the upstream source. Other packages in nixpkgs might expect this to be here.